app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java

```diff
@@ -118,6 +153,51 @@ public void uploadPhoto(View view)
         startActivityForResult(Intent.createChooser(uploadPhotoIntent, getResources().getString(R.string.select_photo)), REQUEST_IMAGE_UPLOAD);
     }
 
+    /**
+     * Deletes the picture associated with the book being edited
+     */
+    private void deletePhoto()
```

Comment:
An execption thrown when:
click edit abook icon with existing image.
click upload image to upload new image. 
click delete image button. 
click save. 
exception thrown in deletePhoto()

will further debug.

Reply:
Thanks for the bug report. I can debug this if you're still working on other things.

Reply:
thanks.
Just this bug. Otherwise looks great!

Reply:
Do you mind rechecking this? I can't replicate this issue.

Reply:
https://drive.google.com/drive/folders/1b8olsnzzNRVxzyfLLsH0dDHr_Wp4KKZf?usp=sharing
I attached a recording
was this the expected behavior? Might be an edge case

Reply:
Damn I forgot to reassign the fragment's bookPhotoDowloadUrl and that's what cause the bug. Thanks for catching this!

Reply:
Great!